src/AutoRest.CSharp/Common/Input/CadlInputEnumTypeConverter.cs

```diff
@@ -73,16 +73,38 @@ public static InputEnumType CreateEnumType(ref Utf8JsonReader reader, string? id
                 throw new JsonException("Enum must have at least one value");
             }
 
-            valueType ??= InputPrimitiveType.Int32;
-            var enumType = new InputEnumType(name, ns, accessibility, deprecated, description, usage, valueType, normalizeValues(allowedValues, valueType), isExtendable);
+            InputPrimitiveType? currentType = null;
+            foreach (var value in allowedValues)
+            {
+                switch (value.Value)
+                {
+                    case int i:
+                        if (currentType == InputPrimitiveType.String)
+                            throw new JsonException($"Enum value types are not consistent.");
```

Comment:
Could you provide more info into the exception for better trouble shooting like:

Enum value types are not consistent, expect: string, actual: integer.  etc